app/src/main/java/com/jvoye/tasky/auth/presentation/register/RegisterState.kt

```diff
@@ -0,0 +1,14 @@
+package com.jvoye.tasky.auth.presentation.register
+
+import androidx.compose.foundation.text.input.TextFieldState
+import com.jvoye.tasky.auth.domain.PasswordValidationState
+
+data class RegisterState(
+    val name: TextFieldState = TextFieldState(),
+    val email: TextFieldState = TextFieldState(),
+    val isNameValid: Boolean = true,
+    val isEmailValid: Boolean = true,
+    val password: TextFieldState = TextFieldState(),
+    val isPasswordVisible: Boolean = false,
+    val passwordValidationState: PasswordValidationState = PasswordValidationState()
+)
```

Comment:
A small issue that can happen with `TextFieldState` being stored in the state data class is that assertions in test cases can break. If you Cmd + Click into the `TextFieldState`, you'll see that it's a normal class which means instances are compared based on reference (and not based on true equality like normal fields in a data class).
An assertion like this would therefore fail in a test case, which you would typically expect to pass:
```kotlin
assertThat(TextFieldState()).isEqualTo(TextFieldState())
```
This is nothing to worry about and not a reason to not use text field states in your data class, but something to be aware about, so that when it happens you know why.